From 2b855a685a8dd445c9e20139967b4c16aab3df12 Mon Sep 17 00:00:00 2001
From: Mani Narayan <mnarayan@cloudera.com>
Date: Fri, 8 Apr 2016 14:57:41 -0700
Subject: [PATCH 1410/2848] CLOUDERA-BUILD: Downloading apache tomcat from
 archive.cloudera.com

archive.apache.org has blacklisted Cloudera's build workers on ec2, and
fetching artifacts from there fails. The artifact that Hadoop relies on is apache-tomcat,
and this has been moved to archive.cloudera.com. This change simply
updates pom.xml to point to archive.cloudera.com

For more information:
https://jira.cloudera.com/browse/RELENG-281.

Testing done:
Generated effetive pom.xml, and verified that the url points to
http://mirror.infra.cloudera.com/apache/tomcat/tomcat-6/v6.0.44/bin/apache-tomcat-6.0.44.tar.gz
Will also build locally to verify that builds work, and update the
review once done.

Change-Id: I6c344da432f453416b41aa84674875f9d26bf653
(cherry picked from commit ba807c8c04b8f000d15540a382a9f7fbed539296)
(cherry picked from commit 02e41f13fcf5cd5cea97491b0af1484ba5de2675)
(cherry picked from commit 8fa90882a2eb14fc105a0eb51f136d72797de289)
---
 hadoop-common-project/hadoop-kms/pom.xml       |    4 +---
 hadoop-hdfs-project/hadoop-hdfs-httpfs/pom.xml |    2 +-
 pom.xml                                        |    1 +
 3 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/hadoop-common-project/hadoop-kms/pom.xml b/hadoop-common-project/hadoop-kms/pom.xml
index 346bef9..a182d06 100644
--- a/hadoop-common-project/hadoop-kms/pom.xml
+++ b/hadoop-common-project/hadoop-kms/pom.xml
@@ -37,9 +37,7 @@
     <kms.tomcat.dist.dir>
       ${project.build.directory}/${project.artifactId}-${project.version}/share/hadoop/kms/tomcat
     </kms.tomcat.dist.dir>
-    <tomcat.download.url>
-      http://archive.apache.org/dist/tomcat/tomcat-6/v${tomcat.version}/bin/apache-tomcat-${tomcat.version}.tar.gz
-    </tomcat.download.url>
+    <tomcat.download.url>${tomcat.mirror}</tomcat.download.url>
   </properties>
 
   <dependencies>
diff --git a/hadoop-hdfs-project/hadoop-hdfs-httpfs/pom.xml b/hadoop-hdfs-project/hadoop-hdfs-httpfs/pom.xml
index 59370a9..b4e3967 100644
--- a/hadoop-hdfs-project/hadoop-hdfs-httpfs/pom.xml
+++ b/hadoop-hdfs-project/hadoop-hdfs-httpfs/pom.xml
@@ -45,7 +45,7 @@
     </httpfs.tomcat.dist.dir>
     <kerberos.realm>LOCALHOST</kerberos.realm>
     <test.exclude.kerberos.test>**/TestHttpFSWithKerberos.java</test.exclude.kerberos.test>
-    <tomcat.download.url>http://archive.apache.org/dist/tomcat/tomcat-6/v${tomcat.version}/bin/apache-tomcat-${tomcat.version}.tar.gz</tomcat.download.url>
+    <tomcat.download.url>${tomcat.mirror}</tomcat.download.url>
   </properties>
 
   <dependencies>
diff --git a/pom.xml b/pom.xml
index 9324cd6..23984f5 100644
--- a/pom.xml
+++ b/pom.xml
@@ -99,6 +99,7 @@ xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xs
     <distMgmtStagingId>apache.staging.https</distMgmtStagingId>
     <distMgmtStagingName>Apache Release Distribution Repository</distMgmtStagingName>
     <distMgmtStagingUrl>https://repository.apache.org/service/local/staging/deploy/maven2</distMgmtStagingUrl>
+    <tomcat.mirror>http://archive.cloudera.com/tarballs/apache/tomcat/tomcat-6/v${tomcat.version}/bin/apache-tomcat-${tomcat.version}.tar.gz</tomcat.mirror>
 
     <!-- platform encoding override -->
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-- 
1.7.9.5

